utf8parse/src/lib.rs

```diff
@@ -98,13 +98,30 @@ mod tests {
     use std::io::prelude::*;
     use std::fs::File;
 
+    // i have no idea what I'm doing here
+    impl Receiver for String {
```

Comment:
What you've got here is nearly correct. The `Receiver` type is called by the `utf8parse::Parser` whenever it completes a codepoint or an invalid sequence. There's an API [`String::push(&mut self, char)`](https://doc.rust-lang.org/std/string/struct.String.html?search=#method.push) which should allow you to build a string using these callbacks.